### PR TITLE
strip quotes from shard iterators before parsing base64

### PR DESCRIPTION
**Pull request by dfangl**
## Changes Introduced

Allow shard iterators to be passed with surrounding double quotes `"`.
This became necessary as AWS itself also allows this.

I stripped the leading and trailing slashes before parsing the base64, and added an integration test making a `get-records` request with double quotes around the shard iterator as well.

I added a test, validated against AWS, to LocalStack as well (see localstack/localstack#9884).

Once this is merged and released, I will bump the `kinesis-mock` version in LocalStack in the same PR, so the test gets green.

## Applicable linked issues

#682 

## Checklist (check all that apply)

- [x] This change maintains backwards compatibility
- [x] I have introduced tests for all new features and changes
- [ ] I have added documentation covering all new features and changes
- [x] This pull-request is ready for review

### CODE REVIEW

File: kinesis-mock/src/main/scala/kinesis/mock/models/ShardIterator.scala [hunk 1]
```diff
@@ -30,7 +30,7 @@ import kinesis.mock.validations.CommonValidations
 
 final case class ShardIterator(value: String) {
   def parse(now: Instant): Response[ShardIteratorParts] = {
-    val decoded = Base64.getDecoder.decode(value)
+    val decoded = Base64.getDecoder.decode(value.replaceAll("^\"|\"$", ""))
```
Comment (etspaceman) [inline kinesis-mock/src/main/scala/kinesis/mock/models/ShardIterator.scala:33]: Are there cases where this could be provided non-escaped quotes?

**Comment by etspaceman** [comment]
Another test that we could potentially add is one that leverages the `awscli` image in the docker-compose setup here:

https://github.com/etspaceman/kinesis-mock/blob/main/docker/docker-compose.yml#L38

You'd create a script to go through the motions of your original issue, and have the awscli image run it.

File: kinesis-mock/src/main/scala/kinesis/mock/models/ShardIterator.scala (hunk 1 above)
Comment (etspaceman) [reply kinesis-mock/src/main/scala/kinesis/mock/models/ShardIterator.scala:33]: Ah, nevermind, misread this

**Comment by dfangl** [comment]
I pushed another commit adding that change. To not spam the docker compose file with executions of the awscli image, I merged it with the current list-streams test, inside a test script, hope this is ok.

Please let me know what you think :)

**Comment by etspaceman** [comment]
Yep that's exactly what I was looking for 😄

**Comment by etspaceman** [comment]
Ah creating a test stream breaks some of the existing tests. Can your script also include deleting the stream?

**Comment by dfangl** [comment]
Deleting the stream now. If this still leads to issues for whatever reason, we might just reuse one of the already initialized streams.

**Review by etspaceman — APPROVED** [review]

**Pull request merged**